packages/wonder-blocks-clickable/components/clickable.js

```diff
@@ -0,0 +1,168 @@
+     * by default if react-router is present
+     */
+    href?: string,
+
+    /**
+     * Styles to apply to the Clickable compoenent
+     */
+    style?: StyleType,
+
+    /**
+     * Disables or enables the child; defaults to false
+     */
+    disabled: boolean,
+
+    /**
+     * The role of the component, can be a role of type ClickableRole
+     */
+    role?: ClickableRole,
+
+    /**
+     * Avoids client-side routing in the presence of the href prop
+     */
+    skipClientNav?: boolean,
+
+    /**
+     * Use to label the component
+     */
+    "aria-label": string,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+|};
+
+const StyledAnchor = addStyle<"a">("a");
+const StyledButton = addStyle<"button">("button");
+const StyledLink = addStyle<typeof Link>(Link);
+
+export default class Clickable extends React.Component<Props> {
+    static contextTypes = {router: PropTypes.any};
+
+    static defaultProps = {
+        disabled: false,
+        "aria-label": "",
+    };
+
+    getCorrectTag = (clickableState: ClickableState, commonProps: mixed) => {
+        const activeHref = this.props.href && !this.props.disabled;
+        const useClient = this.context.router && !this.props.skipClientNav;
+
+        if (activeHref && useClient) {
+            return (
+                <StyledLink
+                    {...commonProps}
+                    to={this.props.href}
+                    aria-disabled={this.props.disabled ? "true" : undefined}
+                >
+                    {this.props.children(clickableState)}
+                </StyledLink>
+            );
+        } else if (activeHref && !useClient) {
+            return (
+                <StyledAnchor
+                    {...commonProps}
+                    href={this.props.href}
+                    aria-disabled={this.props.disabled ? "true" : undefined}
+                >
+                    {this.props.children(clickableState)}
+                </StyledAnchor>
+            );
+        } else {
+            return (
+                <StyledButton
+                    {...commonProps}
+                    type="button"
+                    disabled={this.props.disabled}
+                >
+                    {this.props.children(clickableState)}
+                </StyledButton>
+            );
+        }
+    };
+
+    render() {
+        const ClickableBehavior = getClickableBehavior(
+            this.props.href,
+            this.props.skipClientNav,
+            this.context.router,
+        );
+
+        return (
+            <ClickableBehavior onClick={this.props.onClick}>
+                {(state, handlers) =>
+                    this.getCorrectTag(state, {
+                        // eslint-disable-next-line react/prop-types
+                        "aria-label": this.props["aria-label"],
+                        "data-test-id": this.props.testId,
+                        role: this.props.role,
```

Comment:
This `role` prop shouldn't be part of `commonProps` b/c `<button>` doesn't need to use this role.

More info here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Best_practices

Reply:
thanks for catching that!